src/main/java/platform/codingnomads/co/springweb/resttemplate/DELETE/DeleteMain.java

```diff
@@ -69,6 +69,43 @@ public CommandLineRunner run() throws Exception {
                         + newTask.getId(), HttpMethod.DELETE, httpEntity, ResponseObject.class);
             } catch (HttpClientErrorException e) {
                 System.out.println(e.getMessage());
+            } */
+            UserTemplate userTemplate = UserTemplate.builder().email("bill@bob.com").first_name("Bill").last_name("Bob").build();
+
+            // POST new User to server
+            UserResponseObject responseObject = restTemplate.postForObject("http://demo.codingnomads.co:8080/tasks_api/users/", userTemplate, UserResponseObject.class);
+
+            // Confirm data was returned & avoid NullPointerExceptions
+            if (responseObject == null) {
+                throw new Exception("The server did not return anything. Not even a ResponseObject!");
+            } else if (responseObject.getData() == null) {
+                throw new Exception("The server encountered this error while creating the user:" + responseObject.getError().getMessage());
+            } else {
+                userTemplate = responseObject.getData();
+            }
```

Comment:
Embedded `if` statements are harder to read. To make code less complicated you can use the "early exit" technique. 

Easier to explain with example:
```suggestion
            if (responseObject == null) {
                throw new Exception("The server did not return anything. Not even a ResponseObject!");
            } 
            if (responseObject.getData() == null) {
                throw new Exception("The server encountered this error while creating the user:" + responseObject.getError().getMessage());
            } 
            
            userTemplate = responseObject.getData();
```
Basically: "check criteria -> exit if it is not satisfied". By doing it you guard subsequent code based on the criteria. 
In the example `responseObject == null` check guarantees that once you pass this check the `responseObject` is `not null`. The guarrantie comes from `throw` functionality.

Reply:
Ok, I see. Because the throw function will cause an early cancellation of the current method and return back up to the calling method, right?

By the way, I didn't write any of that code. Still good to know.